synapse/lib/coro.py

```diff
@@ -132,3 +135,42 @@ def genrhelp(f):
     def func(*args, **kwargs):
         return GenrHelp(f(*args, **kwargs))
     return func
+
+def _exectodo(que, todo):
+    func, args, kwargs = todo
+    try:
+        que.put(func(*args, **kwargs))
```

Comment:
Do you want to allow async targets? If so, we need to do some special handling here instead of just calling `que.put()`